mistune_contrib/passthrough.py

```diff
@@ -0,0 +1,234 @@
+            if text:  # pragma: no cover
+                raise RuntimeError('Infinite loop at: %s' % text)
+        return raw_tokens
+
+class RawInlineLexer(InlineLexer):
+    def __init__(self, rules=None, **kwargs):
+        super(RawInlineLexer, self).__init__(Renderer(), rules=rules, **kwargs)
+
+    def parse(self, text, rules=None):
+        if not rules:
+            rules = list(self.default_rules)
+
+        if self._in_footnote and 'footnote' in rules:
+            rules.remove('footnote')
+
+        def manipulate(text):
+            for key in rules:
+                pattern = getattr(self.rules, key)
+                m = pattern.match(text)
+                if not m:
+                    continue
+                return m, key
+            return False  # pragma: no cover
+
+        tokens = []
+
+        self.line_started = False
+        while text:
+            ret = manipulate(text)
+            self.line_started = True
+            if ret is not False:
+                m, rule = ret
+                tokens.append({'type': rule, 'block': False, 'text': m.group(0)})
+                text = text[len(m.group(0)):]
+                continue
+            if text:  # pragma: no cover
+                raise RuntimeError('Infinite loop at: %s' % text)
+
+        return tokens
+
+class MarkdownTransformer(object):
+    def __init__(self, inline=None, block=None, **kwargs):
+        if inline and inspect.isclass(inline):
+            inline = inline(**kwargs)
+        if block and inspect.isclass(block):
+            block = block(**kwargs)
+
+        if inline:
+            self.inline = inline
+        else:
+            self.inline = RawInlineLexer(**kwargs)
+
+        self.block = block or RawBlockLexer()
+        self.footnotes = []
+        self.tokens = []
+
+        # detect if it should parse text in block html
+        self._parse_block_html = kwargs.get('parse_block_html')
+
+    def pop(self):
+        if not self.tokens:
+            return None
+        self.token = self.tokens.pop()
+        return self.token
+
+    def peek(self):
+        if self.tokens:
+            return self.tokens[-1]
+        return None  # pragma: no cover
+
+    def output(self, text, rules=None):
+        self.tokens = self.block(text, rules)
+        self.tokens.reverse()
+
+        self.inline.setup(self.block.def_links, self.block.def_footnotes)
+
+        results = []
+        while self.pop():
+            results.append(self.tok())
+        return ''.join(results)
+
+    def tok(self):
+        t = self.token['type']
+        #print('block', t, repr(self.token['text']))
+
+        # sepcial cases
+        if t.endswith('_start'):
+            t = t[:-6]
+
+        return getattr(self, 'output_%s' % t, self.block_default)()
+
+    def inline_tokens(self, text, rules=None):
+        results = []
+        for t in self.inline.parse(text, rules=rules):
+            meth = getattr(self, 'output_'+t['type'], self.inline_default)
```

Comment:
validate syntax with flake8.

Reply:
Not sure if you wanted a change on this particular line, but I've made flake8 happy.